### PR TITLE
Adding telemetry API documentation

### PR DESCRIPTION
**Pull request by rivkahstandig3636**
For [CH24701](https://app.clubhouse.io/rollbar/story/24701/telemetry-update-api-post-docs)

I'd like to get this reviewed before merging.

### CODE REVIEW

File: api/items_post.md [hunk 1]
```diff
@@ -80,6 +80,41 @@ understands. The meaning of each key is explained in the comments.
...
       // If an iOS crash report, use "crash_report"
 
+      // Optional: "telemetry". Only applicable if you are sending [telemetry data](/docs/notifier/rollbar.js/#telemetry). 
+      "telemetry": {
+        // Required: level
+        // The severity level of the telemetry data. One of: "critical", "error", "warning", "info", "debug".
+        "level": "info",
+
+        // Required: type
+        // The type of telemetry data. One of: "log", "network", "dom", "navigation", "error", "custom".
```
Comment (rokob) [inline api/items_post.md]: `custom` is not a type is it? Rollbar.js uses `manual` for the type of manual generated events

File: api/items_post.md [hunk 2]
```diff
@@ -80,6 +80,41 @@ understands. The meaning of each key is explained in the comments.
...
+        // If type above is "network", body should contain "method", "url", and "status_code" keys.
+        // If type above is "dom", body should contain "element" key.
+        // If type above is "navigation", body should contain "from" and "to" keys.
+        // If type above is "error", body should contain "message" key. 
+        "body": {
+          "subtype": "xhr",
+          "method": "GET",
+          "url": "/api/1/item/4/",
+          "status_code": "200",
+          "start_timestamp_ms": 1500413394557,
```
Comment (rokob) [inline api/items_post.md]: rollbar.js uses `start_time_ms` and `end_time_ms` for these, not sure if that is exactly what the backend requires or not.

Comment (coryvirok) [inline api/items_post.md]: I believe the backend does expect those, yes.

File: api/items_post.md [hunk 3]
```diff
@@ -80,6 +80,41 @@ understands. The meaning of each key is explained in the comments.
...
+        // Required: type
+        // The type of telemetry data. One of: "log", "network", "dom", "navigation", "error", "custom".
+        "type": "network",
+
+        // Required: source
+        // The source of the telemetry data. Usually "client" or "server".
+        "source": "client",
+
+        // Required: timestamp_ms
+        // When this occurred, as a unix timestamp.
```
Comment (coryvirok) [inline api/items_post.md]: I'd add "... as a unix timestamp in milliseconds."

File: api/items_post.md (hunk 1 above)
Comment (rivkahstandig3636) [inline api/items_post.md]: I checked against [this PR](https://github.com/rollbar/moxapi/pull/169) and the [tech spec](https://rollbar.quip.com/n2uFATVRZWF9/Telemetry-Tech-Spec) to write these docs. I'm not sure; do you know where in the code I can find this? I've been looking but haven't found a list of accepted type values.

Comment (rivkahstandig3636) [reply api/items_post.md]: @coryvirok or @rokob can you help me with this?

Comment (coryvirok) [reply api/items_post.md]: Looks like `manual` is indeed used on the frontend however that appears to be a bug based on the techspec. 

@chrisbarmonde can we add `custom` as an alias to `manual`? I think the only place that checks/uses it is in the react code.

https://github.com/rollbar/mox/blob/c069ed3880593b65f7845bcba17fc0fca8ba4d3a/mox/frontend/js/item/telemetry/Breadcrumb.js#L14

Comment (chrisbarmonde) [reply api/items_post.md]: It kinda sounds like what we have now is working fine and it's just the techspec that isn't accurate? While I started off using the spec, I ultimately just followed along with what was happening in the notifier which is why the notifier and the frontend align right now.

**Comment by ezarowny** [comment]
```JavaScript
var telemetrySchemaV4 = {
  type: 'object',
  required: ['level', 'type', 'source', 'timestamp_ms'],
  properties: {
    level: {type: 'string'},
    type: {type: 'string'},
    source: {type: 'string'},
    timestamp_ms: {type: 'integer'},
    body: {type: ['object', 'null']}
  }
};
```

File: api/items_post.md (hunk 1 above)
Comment (coryvirok) [reply api/items_post.md]: Ah, I didn't realize rollbar.js also used 'manual', as Andrew mentioned. (https://github.com/rollbar/rollbar.js/blob/5d3c47f1bb2e944d35b7107e822ab6e2ebab0ac2/src/telemetry.js#L44)

@rivkahstandig3636 let's just change custom -> manual and call it a day.

**Pull request merged**